### PR TITLE
chore(pipeline) use new trusted.ci ZIP credentials to deploy generated artefacts

### PR DESCRIPTION
**Pull request by dduportal**
This PR switches the CD part of this project (running in trusted.ci.jenkins.io) to reuse the same credentials, introduced in https://github.com/jenkins-infra/update-center2/pull/787, as the jenkins-infra/update-center2 job.

It also avoids to repeat credentials between the 2 jobs (SSH key for instance...) and improves the configuration management (we change Rsync user/host/etc only once!)

Notes:

- The `azcopy-redirection` Azure file share won't need to be updated here as it only hosts the Apache `htaccess` files (as far as I can tell we don't have any `.htaccess` files in the `./updates/` directory in production: `find /var/www/updates.jenkins.io/updates/ -type f -name "*.htaccess" | wc -l` says `0`
- There is room for improvement: once we'll introduce new mirror SSH/S3 or other targets, we'll have to update manually the pipeline

~Keeping as draft until we can test this new pipeline on trusted.ci.jenkins.io using a Replay with custom rsync targets~

Update: ready to roll. We've tested by part using replays (with dummy content and only azcopy publication, then only S3/R2 publication) and fully with a successful replay (including the rsync to current production after a carefull backup)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -66,47 +66,75 @@ node('linux') {
...
+                    // TODO: find a way to reuse 'SYNC_UC_TASKS' from https://github.com/jenkins-infra/update-center2/blob/master/site/publish.sh#L9 to avoid repetition and automate delivery.
                 ]) {
+                    // Rsync copy tasks
+                    sh '''
+                    source "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-rsync-updates.jenkins.io"
+
+                    # Required variables that should now be set from the .env file
+                    : "${RSYNC_HOST?}" "${RSYNC_USER?}" "${RSYNC_GROUP?}" "${RSYNC_REMOTE_DIR?}" "${RSYNC_IDENTITY_NAME?}"
+
+                    rsync -rlptDvz -e "ssh -o StrictHostKeyChecking=no -i ${UPDATE_CENTER_FILESHARES_ENV_FILES}/${RSYNC_IDENTITY_NAME}" --exclude=.svn --chown="${RSYNC_USER}":"${RSYNC_GROUP}" updates/ "${RSYNC_USER}"@$"{RSYNC_HOST}":"{RSYNC_REMOTE_DIR}"/updates/
```
Comment (dduportal) [inline Jenkinsfile:81]:
```suggestion
                    rsync -rlptDvz -e "ssh -o StrictHostKeyChecking=no -i ${UPDATE_CENTER_FILESHARES_ENV_FILES}/${RSYNC_IDENTITY_NAME}" --exclude=.svn --chown="${RSYNC_USER}":"${RSYNC_GROUP}" ./updates/ "${RSYNC_USER}"@$"{RSYNC_HOST}":"{RSYNC_REMOTE_DIR}"/updates/
```

File: Jenkinsfile [hunk 2]
```diff
@@ -66,47 +66,75 @@ node('linux') {
...
-                    'UPDATES_R2_ENDPOINT=https://8d1838a43923148c5cee18ccc356a594.r2.cloudflarestorage.com',
-                    'STORAGE_FILESHARE=updates-jenkins-io',
-                    'STORAGE_NAME=updatesjenkinsio',
-                    'STORAGE_DURATION_IN_MINUTE=5',
-                    'STORAGE_PERMISSIONS=dlrw'
+                    // TODO: find a way to reuse 'SYNC_UC_TASKS' from https://github.com/jenkins-infra/update-center2/blob/master/site/publish.sh#L9 to avoid repetition and automate delivery.
                 ]) {
+                    // Rsync copy tasks
+                    sh '''
+                    source "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-rsync-updates.jenkins.io"
```
Comment (dduportal) [inline Jenkinsfile:76]:
```suggestion
                    . "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-rsync-updates.jenkins.io"
```

File: Jenkinsfile [hunk 3]
```diff
@@ -66,47 +66,75 @@ node('linux') {
...
+
+                    # Required variables that should now be set from the .env file
+                    : "${RSYNC_HOST?}" "${RSYNC_USER?}" "${RSYNC_GROUP?}" "${RSYNC_REMOTE_DIR?}" "${RSYNC_IDENTITY_NAME?}"
+
+                    rsync -rlptDvz -e "ssh -o StrictHostKeyChecking=no -i ${UPDATE_CENTER_FILESHARES_ENV_FILES}/${RSYNC_IDENTITY_NAME}" --exclude=.svn --chown="${RSYNC_USER}":"${RSYNC_GROUP}" updates/ "${RSYNC_USER}"@$"{RSYNC_HOST}":"{RSYNC_REMOTE_DIR}"/updates/
+                    '''
+
+                    // Azure copy tasks
                     sh '''
+                    source "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-azsync-content"
```
Comment (dduportal) [inline Jenkinsfile:86]:
```suggestion
                    . "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-azsync-content"
```

File: Jenkinsfile [hunk 4]
```diff
@@ -66,47 +66,75 @@ node('linux') {
...
+                        --skip-version-check `# Do not check for new azcopy versions (we have updatecli for this)` \
                         --exclude-path '.svn' \
                         --recursive=true \
+                        --delete-destination=true \
                         ./updates/ "${fileShareSignedUrl}"
+                    '''
+
+                    // AWS copy tasks
+                    sh '''
+                    source "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-s3sync-westeurope"
```
Comment (dduportal) [inline Jenkinsfile:110]:
```suggestion
                    . "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-s3sync-westeurope"
```

File: Jenkinsfile [hunk 5]
```diff
@@ -66,47 +66,75 @@ node('linux') {
...
+                    aws s3 sync ./updates/ s3://"${BUCKET_NAME}"/updates/ \
+                        --no-progress \
+                        --no-follow-symlinks \
+                        --size-only \
+                        --exclude '.svn' \
+                        --exclude '.htaccess' \
+                        --endpoint-url "${BUCKET_ENDPOINT_URL}"
+                    '''
+                    sh '''
+                    source "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-s3sync-eastamerica"
```
Comment (dduportal) [inline Jenkinsfile:125]:
```suggestion
                    . "${UPDATE_CENTER_FILESHARES_ENV_FILES}/.env-s3sync-eastamerica"
```

**Comment by dduportal** [comment]
Yuu can check the trusted replay build at https://trusted.ci.jenkins.io:1443/job/crawler/14983/ (to be unlocked once this PR is merged)

**Review by MarkEWaite — APPROVED** [review]

**Review by timja — COMMENTED** [review]
I would recommend moving these inline scripts into external scripts

you'll get syntax highlighting, easier to run them to debug them and e.g inline shellcheck in IDEs

**Comment by dduportal** [comment]
> I would recommend moving these inline scripts into external scripts
> 
> you'll get syntax highlighting, easier to run them to debug them and e.g inline shellcheck in IDEs

Absolutely, as a second step: the ability to test the "real life CD" using replay was the trigger to have them inline at first.

**Comment by dduportal** [comment]
Update:

- The builds on both ci.jio (CI) and trusted.ci (CD were successful after this PR
- The impact on publication time is the following: went from ~10s to ~1m => room for improvement (most probably once the publication process will be factorized with update-center2 which uses GNU `parallel`)

**Comment by dduportal** [comment]
> Probably not such an issue on this repo but it will be consuming cloud credits so faster is always better there.

+100. The real problem is around maintenance: the PR here improved a bit as we only have one set of credential for both update-center2 and crawler. But using the same method for publication would avoid maintenance pains in the future.

It's just that we proceed tiny steps by steps as this is a sensitive service

**Comment by dduportal** [comment]
> I would recommend moving these inline scripts into external scripts
> 
> you'll get syntax highlighting, easier to run them to debug them and e.g inline shellcheck in IDEs

Btw: https://github.com/jenkins-infra/crawler/issues/149

**Pull request merged**